controller.py

```diff
@@ -81,46 +83,157 @@
 
 OPDS_2_MEDIA_TYPE = 'application/opds+json'
 
+class MetadataWrangler(object):
+    """The metadata wrangler itself.
 
-def authenticated_client_from_request(_db, required=True):
-    header = request.headers.get('Authorization')
-    if header and 'bearer' in header.lower():
-        shared_secret = base64.b64decode(header.split(' ')[1])
-        client = IntegrationClient.authenticate(_db, shared_secret)
-        if client:
-            return client
-    if not required and not header:
-        # In the case that authentication is not required
-        # (i.e. URN lookup) return None instead of an error.
-        return None
-    return INVALID_CREDENTIALS
```

Comment:
This is implemented in a more robust form as a class method of MetadataWrangler. That makes it easy to swap in a mocked implementation when we're testing the @accepts_auth and @requires_auth annotators.